[stable/ark] Several improvements on naming, decoupling and metrics

PabloCastellano opened the pull request:
Ping @agolomoodysaada who was the last person who updated this chart

#### What this PR does / why we need it:

* Use new Ark naming:
    * Rename backupStorageProvider to backupStorageLocation
    * Rename persistentVolumeProvider to volumeSnapshotLocation
* Add configuration.provider so that backupStorageLocation is decoupled from
  the cloud provider where ark it deployed to (hence we can have multiple backup locations in different cloud providers)
* Fixed typo in metricsAddress (now renamed to configuration.metrics.address)
* Expose metrics port with configuration.metrics.enabled
* Support new prefix parameter for backupStorageLocation

#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

Commit:
[stable/ark] Several improvements on naming, decoupling and metrics

* Use new naming:
    * Rename backupStorageProvider to backupStorageLocation
    * Rename persistentVolumeProvider to volumeSnapshotLocation
* Add configuration.provider so that backupStorageLocation is decoupled from
  the cloud provider where ark it deployed to
* Fixed typo in metricsAddress (now renamed to configuration.metrics.address)
* Expose metrics port with configuration.metrics.enabled

Signed-off-by: Pablo Castellano <pablo.castellano@schibsted.com>

agolomoodysaada reviewed (approved):
LGTM! Thank you  🎉

Commit:
Allow setting new prefix parameter

Signed-off-by: Pablo Castellano <pablo.castellano@schibsted.com>

Commit:
Fix and document prefix parameter

Signed-off-by: Pablo Castellano <pablo.castellano@schibsted.com>

unguiculus commented on stable/ark/Chart.yaml:
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 description: A Helm chart for ark
 name: ark
-version: 3.0.0
+version: 3.0.1
```
Seems like this is a breaking change which needs a major version bump.

PabloCastellano replied:
Agree, changed!

Commit:
Major version bump

Signed-off-by: Pablo Castellano <pablo.castellano@schibsted.com>

PabloCastellano committed:
Add prometheus monitoring

Originally by JoschaLaubach (https://github.com/helm/charts/pull/9496)
Adapted by PabloCastellano to latest chart available

Signed-off-by: Pablo Castellano <pablo@anche.no>

PabloCastellano committed:
Merge remote-tracking branch 'upstream/master' into ark
